salt/proxy/restconf.py

```diff
@@ -0,0 +1,237 @@
+
+Usage
+=====
+
+.. note::
+
+    To be able to use this module you need to enable RESTCONF on your device
+    and have https enabled.
+
+    Cisco Configuration example:
+
+    .. code-block:: bash
+
+        switch# conf t
+        switch(config)# restconf
+        switch(config)# ip http secure-server
+
+.. note::
+
+    RESTCONF requires modern OS distributions.
+    This plugin has been written specifically to use JSON RESTCONF endpoints
+
+Pillar
+------
+
+The ``restconf`` proxy configuration requires the following parameters in order
+to connect to the network switch:
+
+transport: ``https`` (str)
+    Specifies the type of connection transport to use. Valid values for the
+    connection are ``https``, and  ``http``.
+    The RESTCONF standard explicitly requires https, but http is included as an option
+    as some manufacturers have ignored this requirement.
+
+hostname: (str)
+    The IP address or DNS host name of the RESTCONF device.
+
+username: (str)
+    The username for the device to authenticate the RESTCONF requests.
+
+password: (str)
+    The password for the device to authenticate the RESTCONF requests.
+
+verify: ``True`` or ``False`` (str, optional, default:true)
+    Verify the RESTCONF SSL certificate?
+
+    When there is no certificate configuration on the device and this option is
+    set as ``True`` (default), the commands will fail with the following error:
+    ``SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed``.
+
+    .. warning::
+      In this case, you either need to configure a proper certificate on the
+      device (*recommended*), or bypass the checks setting this argument as ``False``
+      with all the security risks considered as you may be MITM'd.
+
+Proxy Pillar Example
+--------------------
+
+.. code-block:: yaml
+
+    proxy:
+      proxytype: restconf
+      host: switch1.example.com
+      username: example
+      password: example
+      verify: false
+"""
+
+
+import copy
+import json
+import logging
+
+import salt.utils.http
+
+# Import Salt modules
+from salt.exceptions import SaltException
+
+# -----------------------------------------------------------------------------
+# proxy properties
+# -----------------------------------------------------------------------------
+
+__proxyenabled__ = ["restconf"]
+# proxy name
+
+# -----------------------------------------------------------------------------
+# globals
+# -----------------------------------------------------------------------------
+
+__virtualname__ = "restconf"
+log = logging.getLogger(__file__)
+restconf_device = {}
+
+# -----------------------------------------------------------------------------
+# property functions
+# -----------------------------------------------------------------------------
+
+
+def __virtual__():
+    """
+    This Proxy Module is widely available as there are no external dependencies.
+    """
+    log.debug("restconf proxy __virtual__() called...")
+    return __virtualname__
```

Comment:
```suggestion
    return __virtualname__
```

Reply:
`__virtual__()` gets called a lot, and that log message will quickly turn into log spamming.